toolkit/tools/pkg/imagecustomizerlib/kernelmoduleutils.go

```diff
@@ -0,0 +1,313 @@
+			logger.Log.Infof("Disabling module %s", moduleName)
+		}
+	}
+
+	if needUpdate {
+		err = os.WriteFile(moduleDisableFilePath, []byte(updatedContent), 0644)
+		if err != nil {
+			return fmt.Errorf("failed to update disable configuration: %w", err)
+		}
+	}
+
+	return nil
+}
+
+func isModuleDisabled(moduleName, moduleDisableFilePath string) (bool, error) {
+	_, err := os.Stat(moduleDisableFilePath)
+	if os.IsNotExist(err) {
+		// File doesn't exist, treat as not disabled.
+		return false, nil
+	} else if err != nil {
+		return false, err
+	}
+
+	content, err := os.ReadFile(moduleDisableFilePath)
+	if err != nil {
+		logger.Log.Errorf("Failed to scan file %s with error %s", moduleDisableFilePath, err)
+		return false, err
+	}
+
+	if strings.Contains(string(content), moduleName) {
+		return true, nil
+	}
+
+	return false, nil
+}
+
+func removeModuleFromDisableList(moduleName, moduleDisableFilePath string) error {
+	disabled, err := isModuleDisabled(moduleName, moduleDisableFilePath)
+	if err != nil {
+		logger.Log.Infof("Error checking %s", moduleDisableFilePath)
+		return err
+	}
+
+	if disabled {
+		logger.Log.Infof("Module %s found in the disabled list. Removing...", moduleName)
+		lines, err := os.ReadFile(moduleDisableFilePath)
+		if err != nil {
+			return fmt.Errorf("failed to write module disable configuration: %w", err)
+		}
+
+		// Filter out the line that contains the module name.
+		var updatedLines []string
+		for _, line := range strings.Split(string(lines), "\n") {
+			if !strings.Contains(line, moduleName) {
+				updatedLines = append(updatedLines, line)
+			}
+		}
+
+		return os.WriteFile(moduleDisableFilePath, []byte(strings.Join(updatedLines, "\n")), 0644)
+	}
+
+	return nil
+}
+
+func updateModulesOptions(moduleOptionsUpdates map[string]map[string]string, moduleOptionsFilePath string) error {
+	optionsToAppend := []string{}
+	var err error
+
+	if len(moduleOptionsUpdates) > 0 {
+		for moduleName, options := range moduleOptionsUpdates {
+			for key, value := range options {
+				logger.Log.Infof("Writing module options %s=%s for module %s", key, value, moduleName)
+				optionsToAppend, err = aggregateModuleOptions(optionsToAppend, moduleOptionsFilePath, moduleName, key, value)
+				if err != nil {
+					return fmt.Errorf("failed to append module option for module %s: %w", moduleName, err)
+				}
+			}
+		}
+
+		file, err := os.OpenFile(moduleOptionsFilePath, os.O_WRONLY|os.O_CREATE|os.O_APPEND, 0644)
+		if err != nil {
+			return fmt.Errorf("failed to open %s: %w", moduleOptionsFilePath, err)
+		}
+		defer file.Close()
+
+		content := strings.Join(optionsToAppend, "\n") + "\n"
+		if _, err = file.WriteString(content); err != nil {
+			return fmt.Errorf("failed to write module options to %s: %w", moduleOptionsFilePath, err)
+		}
+	}
+
+	return nil
+}
+
+func aggregateModuleOptions(aggregatedOptions []string, moduleOptionsFilePath string, moduleName string, optionKey string, optionValue string) ([]string, error) {
```

Comment:
The behavior of this function seems very strange. Why not read the `moduleOptionsFilePath` file in once, do all the changes required, and then write out the file once?